docs/contributing/coding-standards/js.md

```diff
@@ -5,25 +5,52 @@
 JavaScript files have the same name as the component's folder name. Test files have a `.test` suffix placed before the file extension.
 
 ```
-checkboxes
-├── checkboxes.mjs
-└── checkboxes.test.js
+component
+├── component.mjs
+└── component.test.js
 ```
 
 ## Skeleton
 
 ```js
-import { nodeListForEach } from '../vendor/common.mjs'
+/**
+ * Component name
+ *
+ * @class
+ * @param {HTMLElement | Element} $module - HTML element to use for component
+ * @this {Example}
+ */
+function Example ($module) {
+  if (!($module instanceof HTMLElement)) {
+    // Return instance for method chaining
+    // using `new Example($module).init()`
+    return this
```

Comment:
Ahh sorry, I thought it was the preferred newer approach?

[**button.mjs**#L18](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/button/button.mjs#L18)
[**character-count.mjs**#L58](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/character-count/character-count.mjs#L58)
[**error-summary.mjs**#LL27](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/error-summary/error-summary.mjs#LL27)

I picked what looked like the most sensible constructor + init pattern

Happy to pick another flavour. Let me know what you'd prefer to do

Reply:
@romaricpascal it looks like you might have added the `return this` as [part of `08abe72`](https://github.com/alphagov/govuk-frontend/commit/08abe72278ed9cc9a50da78b17c8c37b0e204a0a#diff-556827c86f23bc608ce35469937f49630bc5d893ef8f461bd1566d26deb5eb98R26-R28) – is this something we need to be careful of with ES6 classes?

MDN just says:

> If the constructor function returns a [non-primitive](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#primitive_values), this return value becomes the result of the whole new expression. Otherwise, if the constructor function doesn't return anything or returns a primitive, newInstance is returned instead.

Reply:
> @romaricpascal it looks like you might have added the `return this` as [part of `08abe72`](https://github.com/alphagov/govuk-frontend/commit/08abe72278ed9cc9a50da78b17c8c37b0e204a0a#diff-556827c86f23bc608ce35469937f49630bc5d893ef8f461bd1566d26deb5eb98R26-R28) – is this something we need to be careful of with ES6 classes?

I just hadn't dug deep enough to realise that `undefined` was actually fine as a return value, sorry. I was mostly aware that return values matered in ES6 (as [it's an opportunity to do cool stuff with `Proxy`s by returning them in the constructor instead of letting things through](https://stackoverflow.com/a/40714458)). I prefered to err on the cautious side at the time.

If it makes old Safari happy as well, though, sounds like we should keep it (until we move to ES classes) 😄

Reply:
> @romaricpascal it looks like you might have added the `return this` as [part of `08abe72`](https://github.com/alphagov/govuk-frontend/commit/08abe72278ed9cc9a50da78b17c8c37b0e204a0a#diff-556827c86f23bc608ce35469937f49630bc5d893ef8f461bd1566d26deb5eb98R26-R28) – is this something we need to be careful of with ES6 classes?

@36degrees Any last thoughts on `return` versus `return this`?

Reply:
Not really… I'd be tempted to try and maintain the status quo as much as possible as I think it's likely we're going to make changes to this in v5 anyway. No point sweating things like this if we can potentially make the problem go away entirely in the near future 🪄

Reply:
You might have noticed I've pushed up a new version

I've kept `return this` for `new Component()` as per these examples:

[button.mjs#L20](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/button/button.mjs#L20)
[character-count.mjs#L61](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/character-count/character-count.mjs#L61)
[error-summary.mjs#L29](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/error-summary/error-summary.mjs#L29)

But for `.init()` I've removed all trace of `return this` and kept things as-is

Also removed it from the [existing (incorrect) example](https://github.com/alphagov/govuk-frontend/commit/99fdcf9d81b718bbe68d4d58348d09897f974fba#diff-dbe5a1a904146fcbe119150cba9b3d123aa79d43ff5b2b120ddfd353568164c3R84):

~`var myExample = new Example().init()`~
`var myExample = new Example()`